plugins/deep-link/src/lib.rs

```diff
@@ -105,10 +154,164 @@ mod imp {
+                        file.write_all(
+                            format!(
+                                include_str!("template.desktop"),
+                                name = self.app.config().identifier,
+                                exec = exec
+                            )
+                            .as_bytes(),
+                        )?;
+
+                        Command::new("update-desktop-database")
+                            .arg(target)
+                            .status()?;
+                    }
+
+                    Command::new("xdg-mime")
+                        .args(["default", &file_name, _protocol.as_ref()])
+                        .status()?;
+                }
+            }
+
+            Ok(())
+        }
+
+        /// Unregister the app as the default handler for the specified protocol.
+        ///
+        /// - `protocol`: The name of the protocol without `://`.
+        ///
+        /// ## Platform-specific:
+        ///
+        /// - **macOS / Linux / Android / iOS**: Unsupported.
+        pub fn unregister<S: AsRef<str>>(&self, _protocol: S) -> crate::Result<()> {
+            #[cfg(windows)]
+            {
+                let hkcu = RegKey::predef(HKEY_CURRENT_USER);
+                let base = Path::new("Software")
+                    .join("Classes")
+                    .join(_protocol.as_ref());
+
+                hkcu.delete_subkey_all(base)?;
+            }
+
+            Ok(())
+        }
+
+        /// Check whether the app is the default handler for the specified protocol.
+        ///
+        /// - `protocol`: The name of the protocol without `://`.
+        ///
+        /// ## Platform-specific:
+        ///
+        /// - **macOS / Android / iOS**: Unsupported, always returns `Ok(true)`.
+        pub fn is_registered<S: AsRef<str>>(&self, _protocol: S) -> crate::Result<bool> {
+            #[cfg(windows)]
+            {
+                let hkcu = RegKey::predef(HKEY_CURRENT_USER);
+
+                let cmd_reg = hkcu.open_subkey(format!(
+                    "Software\\Classes\\{}\\shell\\open\\command",
+                    _protocol.as_ref()
+                ))?;
+
+                let registered_cmd: String = cmd_reg.get_value("")?;
+
+                let exe = tauri::utils::platform::current_exe()?
+                    .display()
+                    .to_string()
+                    .replace("\\\\?\\", "");
+
+                Ok(registered_cmd == format!("{} \"%1\"", &exe))
+            }
+            #[cfg(target_os = "linux")]
+            {
+                let file_name = format!(
+                    "{}.desktop",
+                    tauri::utils::platform::current_exe()?
+                        .file_name()
+                        .unwrap()
+                        .to_string_lossy()
+                );
+
+                let output = Command::new("xdg-mime")
+                    .args(["default", &file_name, _protocol.as_ref()])
```

Comment:
shouldn't this be `query` instead of `default`?

Reply:
maybeee, the pr is too old to remember, i'll check

Reply:
i don't know what the hell i did but the whole linux implementation is wrong. One would imagine that i'd be able to copy paste my own code from the v1 plugin into this _but apparently not_ 😡